pandas/tests/series/methods/test_replace.py

```diff
@@ -241,6 +241,13 @@ def test_replace2(self):
         assert (ser[6:10] == -1).all()
         assert (ser[20:30] == -1).all()
 
+    def test_replace_with_dictlike_and_string_dtype(self):
+        # GH 32621
+        s = pd.Series(["one", "two", np.nan], dtype="string")
+        expected = pd.Series(["1", "2", np.nan])
```

Comment:
The expected result here should also be `dtype="string"` I think

Reply:
Apparently that's not the case (even the current version):
```python
>>> import pandas as pd
>>> pd.__version__
'1.0.3'
>>> s = pd.Series(["one", "two"], dtype="string")
>>> expected = pd.Series(["1", "2"], dtype="string")
>>> result = s.replace(to_replace={"one": "1", "two": "2"})
>>> expected
0    1
1    2
dtype: string
>>> result
0    1
1    2
dtype: object
```
I'm not sure if that's to be expected or it should be tackled within a new issue. What do you think?

Reply:
I would say that it is certainly expected for `replace` to again return a string dtype. 

Although of course, there are not guarantees that your replacement value is a string ... 
And given that it is also not working on master, not sure it needs to be fixed here.

For some cases of `replace` it does work fine:

```
In [33]: s = pd.Series(["one", "two"], dtype="string")  

In [34]: s.replace("one", "1")   
Out[34]: 
0      1
1    two
dtype: string
```

Reply:
When not using the argument `to_replace` (i.e. `s.replace("one", "1")`) then the trace is different. It doesn't use the same function to apply the replacement in comparison to `s.replace(to_replace={"one": "1"})`.

Reply:
@chrispe92 can you open an issue for this